crates/sequencer/src/sequencer.rs

```diff
@@ -1016,19 +1013,34 @@ where
         &self,
         pg_connector: PostgresConnector,
     ) -> Result<(), anyhow::Error> {
-        let ledger_commitment_l1_height =
-            self.ledger_db.get_last_sequencer_commitment_l1_height()?;
+        let ledger_commitment_l1_height = self
+            .ledger_db
+            .get_last_sequencer_commitment_l1_height()?
+            .ok_or(anyhow!("No commitment exists"))?;
+        let ledger_commitment = match self
```

Comment:
it would be nice not to create a huge expr inside match. Just IMO

Reply:
I disagree, there's a reason chaining calls is a thing.

Reply:
Right, chaining is fine. I am saying IMO: `let ledger_commitment = CHAINCALL; let ledger_commitment = match ledger_commitment {}` looks better.